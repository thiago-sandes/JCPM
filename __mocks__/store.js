import {createStore} from 'redux';

import allReducers from '../src/redux/reducers/rootReducer';

const store = createStore(allReducers);

export default store;
