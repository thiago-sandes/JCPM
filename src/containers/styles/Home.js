import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    flex: 1,
  },
  error: {
    justifyContent: 'center',
    alignSelf: 'center',
  },
  list: {
    display: 'flex',
  },
  innerContainer: {
    display: 'flex',
    backgroundColor: '#fff',
    flexDirection: 'column',
    alignItems: 'stretch',
    marginTop: '5%',
  },
  flatListContainer: {
    paddingHorizontal: '2%',
    paddingBottom: '50%',
  },
});
