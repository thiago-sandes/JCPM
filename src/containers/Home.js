import React, {useEffect, useState} from 'react';
import {
  FlatList,
  SafeAreaView,
  ScrollView,
  View,
  Text,
  StatusBar,
} from 'react-native';

import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

// Actions
import {setDataAction} from '../redux/actions/dataAction';

// Styles
import styles from './styles/Home';

// Components
import SimpleSplash from '../components/SimpleSplash';
import Card from '../components/Card';

// API
import {getData} from '../api/api';

const Home = (props) => {
  const {data, navigation, setMovies} = props;

  const [splash, setSplash] = useState(true);
  const [searchTerm, setSearchTerm] = useState('Avenger');

  // Handle api call
  const handleApiCall = async () => {
    const response = await getData({s: searchTerm});

    if (response.Search != null) {
      setTimeout(() => {
        setMovies(response.Search);
        setSplash(false);
      }, 4000);
    }
  };

  const onSelectItem = () => {
    console.log('ok');
  };

  useEffect(() => {
    handleApiCall();
  }, [searchTerm]);

  return (
    <View style={styles.container}>
      <StatusBar translucent backgroundColor="transparent" />

      {splash ? (
        <SimpleSplash />
      ) : data.movies ? (
        <ScrollView>
          <FlatList
            data={data.movies}
            keyExtractor={(item) => item.imdbID}
            renderItem={({item}) => (
              <Card
                title={item.Title}
                year={item.Year}
                img={item.Poster}
                id={item.imdbID}
                onSelectResult={onSelectItem}
              />
            )}
          />
        </ScrollView>
      ) : (
        <View style={styles.error}>
          <Text>No results found</Text>
        </View>
      )}
    </View>
  );
};

const mapStateToProps = (state) => {
  const {dataReducer} = state;

  return {
    data: dataReducer,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setMovies: bindActionCreators(setDataAction, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
