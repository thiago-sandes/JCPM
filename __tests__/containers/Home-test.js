import React from 'react';
import renderer from 'react-test-renderer';
import store from '../../__mocks__/MockedStore';
import Home from '../../src/containers/Home';

test('Home snapshot', () => {
  const tree = renderer.create(<Home store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
});
