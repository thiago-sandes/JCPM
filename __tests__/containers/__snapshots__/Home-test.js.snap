// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "orange",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 40,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Carregando os dados...
    </Text>
    <ActivityIndicator
      animating={true}
      color="#fff"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "marginTop": "2%",
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
